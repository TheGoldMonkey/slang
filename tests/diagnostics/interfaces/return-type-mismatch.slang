// return-type-mismatch.slang

//DIAGNOSTIC_TEST:SIMPLE:

enum LobeTypes : uint32_t { None = 0, Diffuse = 1 }

interface IMaterialInstance {
    LobeTypes get_lobe_types();
}

struct Broken : IMaterialInstance {
    uint get_lobe_types() { return LobeTypes::Diffuse; }
}

int test()
{
    Broken b;
    return 0;
}