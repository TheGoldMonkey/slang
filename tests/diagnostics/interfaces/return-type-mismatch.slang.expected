result code = -1
standard error = {
tests/diagnostics/interfaces/return-type-mismatch.slang(12): error 38105: member 'get_lobe_types' does not match interface requirement.
    uint get_lobe_types() { return LobeTypes::Diffuse; }
         ^~~~~~~~~~~~~~
error 38106: member 'get_lobe_types' return type 'uint' does not match interface requirement return type 'LobeTypes'.
note: see interface requirement declaration of 'get_lobe_types'
tests/diagnostics/interfaces/return-type-mismatch.slang(8): note: see interface requirement declaration of 'get_lobe_types'
    LobeTypes get_lobe_types();
              ^~~~~~~~~~~~~~
}
standard output = {
}